quandl/connection.py

```diff
@@ -65,21 +65,17 @@ def handle_api_error(cls, resp):
         if prog.match(code):
             code_letter = prog.match(code).group(1)
 
-        if code_letter == 'L':
-            klass = LimitExceededError
-        elif code_letter == 'M':
-            klass = InternalServerError
-        elif code_letter == 'A':
-            klass = AuthenticationError
-        elif code_letter == 'P':
-            klass = ForbiddenError
-        elif code_letter == 'S':
-            klass = InvalidRequestError
-        elif code_letter == 'C':
-            klass = NotFoundError
-        elif code_letter == 'X':
-            klass = ServiceUnavailableError
-        else:
-            klass = QuandlError
+        d_klass = {
+            'L': LimitExceededError,
+            'M': InternalServerError,
+            'A': AuthenticationError,
+            'P': ForbiddenError,
+            'S': InvalidRequestError,
+            'C': NotFoundError,
+            'X': ServiceUnavailableError
+        }
+        klass = QuandlError
+        if code_letter in d_klass.keys():
```

Comment:
also we might be able to get rid of this `if` statement, and use this instead:
http://stackoverflow.com/questions/9358983/python-dictionary-and-default-values

Dictionaries can accept default values. Just a suggestion, no need to change if we prefer the current PR's style